test/test_main.py

```diff
@@ -11,8 +12,36 @@
     sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
     from main import run
 
-test_files = glob.glob(os.path.join(os.path.dirname(__file__), "files/**/*.ifc"), recursive=True)
-@pytest.mark.parametrize("filename", test_files)
+rule_code_pattern = re.compile(r"^[a-zA-Z]{3}\d{3}$")
+rule_codes = list(filter(lambda arg: rule_code_pattern.match(arg), sys.argv[1:]))
+
+def get_test_files():
+    """
+    Option -> Example
+    Test files for a rule -> 'python3 test_main.py alb001'
+    Also applies for multiple rules -> 'python3 test_main.py alb001 alb002'
+    Test files for a single file -> 'python3 test_main.py <path>.ifc
+    Also applies for multiple files -> 'python3 test_main.py <path1>.ifc <path2>.ifc'
+    Codes and rules can also be combined -> 'python3 test_main.py alb001 <path>.ifc'
+    """
+    rule_code_pattern = re.compile(r"^[a-zA-Z]{3}\d{3}$")
+    rule_codes = list(filter(lambda arg: rule_code_pattern.match(arg), sys.argv[1:]))
```

Comment:
Instead of blindly taking sys.argv I would filter out all the "flags" e.g `-sv` 

```suggestion
    rule_codes = list(filter(lambda arg: rule_code_pattern.match(arg), [a for a in sys.argv[1:] if not a.startswith('-')]))
```